src/components/new-safe/load/logic/index.ts

```diff
@@ -0,0 +1,43 @@
+import type { SafeInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import type { AppDispatch } from '@/store'
+import { addOrUpdateSafe } from '@/store/addedSafesSlice'
+import { upsertAddressBookEntry } from '@/store/addressBookSlice'
+import { defaultSafeInfo } from '@/store/safeInfoSlice'
+
+export const addSafeToWatchlist = (dispatch: AppDispatch, safe: SafeInfo, safeName: string) => {
```

Comment:
Ideally, I would like to reuse this for the add safe form but we don't have access to the `SafeInfo` there. We could create a function to build the `SafeInfo` out of the parameters though.